R/knitr-engine.R

```diff
@@ -180,6 +182,18 @@ eng_python_initialize <- function(options, context, envir) {
   if (is.character(options$engine.path))
     use_python(options$engine.path[[1]])
   
+  if (options$cache > 0) {
+    module <- tryCatch(import("dill"), error = identity)
+    if (inherits(module, "error")) {
+      if (module$message == "ImportError: No module named dill") {
+        warning("The Python module dill was not found. This module is needed for full cache functionality.")
+      } else {
+        stop(module$message)
+      }
+    } else {
+      py_run_string("import dill")
```

Comment:
Is this line necessary? IIUC the above import attempt should ensure the module is loaded if needed. Or is this done to put `dill` into the main module?

Reply:
Nope, it isn't necessary. This was a hold over from a previous caching attempt. I just added a commit removes it. The new commit passes all tests on my end.